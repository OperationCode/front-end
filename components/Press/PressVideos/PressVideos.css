.pressVideos {
  padding-top: 15px;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
}

.pressVideos iframe {
  padding: 25px;
}

/* Video iFrame Media Queries */
@media screen and (max-width: 960px) {
  .pressVideos iframe {
    width: 100%;
    height: 375px;
  }
}

@media screen and (max-width: 720px) {
  .pressVideos iframe {
    height: 250px;
  }
}

@media screen and (max-width: 480px) {
  .pressVideos iframe {
    height: 187.5px;
  }
}
