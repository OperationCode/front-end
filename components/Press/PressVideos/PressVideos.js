import React from 'react';
import styles from './PressVideos.module.css';

function PressVideos() {
  return (
    <div className={styles.pressVideos}>
      <iframe
        title="GitHub Universe: Operation Code"
        src="https://www.youtube.com/embed/xN7yMoe38xc"
        frameBorder="0"
        allowFullScreen
        width="755"
        height="425"
      />
      <iframe
        title="GitHub Universe: Operation Code"
        src="https://player.vimeo.com/video/124866675?byline=0&portrait=0"
        frameBorder="0"
        allowFullScreen
        width="755"
        height="425"
      />
    </div>
  );
}

export default PressVideos;
