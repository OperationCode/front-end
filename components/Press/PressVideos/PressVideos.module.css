.pressVideos {
  padding-top: 15px;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
}

.pressVideos iframe {
  padding: 25px;
}

/* Video iFrame Media Queries */
@media screen and (--large-viewport) {
  .pressVideos iframe {
    width: 100%;
    height: 375px;
  }
}

@media screen and (--medium-viewport) {
  .pressVideos iframe {
    height: 250px;
  }
}

@media screen and (--small-viewport) {
  .pressVideos iframe {
    height: 187.5px;
  }
}
