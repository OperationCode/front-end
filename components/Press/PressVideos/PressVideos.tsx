const pressVideos = [
  { title: 'Github Universe: Operation Code', src: 'https://www.youtube.com/embed/xN7yMoe38xc' },
  {
    title: 'Operation Code - On a Mission to Expand the GI Bill',
    src: 'https://player.vimeo.com/video/124866675?byline=0&portrait=0',
  },
];

function PressVideos() {
  return (
    <div className="flex items-center justify-around flex-wrap gap-5">
      {pressVideos.map(video => (
        <iframe
          key={video.title}
          title={video.title}
          src={video.src}
          frameBorder="0"
          allowFullScreen
          className="aspect-video h-48 md:h-64 lg:h-96"
        />
      ))}
    </div>
  );
}

export default PressVideos;
