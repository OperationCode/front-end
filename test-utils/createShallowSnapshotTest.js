import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
