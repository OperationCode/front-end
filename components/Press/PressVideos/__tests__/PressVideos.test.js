import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import PressVideos from '../PressVideos';

describe('PressVideos', () => {
  test('it should render properly no props', () => createShallowSnapshotTest(<PressVideos />));
});
