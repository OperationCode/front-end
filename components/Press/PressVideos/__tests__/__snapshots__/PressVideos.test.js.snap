// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressVideos it should render properly no props 1`] = `
<div
  className="pressVideos"
>
  <iframe
    allowFullScreen={true}
    frameBorder="0"
    height="425"
    src="https://www.youtube.com/embed/xN7yMoe38xc"
    title="GitHub Universe: Operation Code"
    width="755"
  />
  <iframe
    allowFullScreen={true}
    frameBorder="0"
    height="425"
    src="https://player.vimeo.com/video/124866675?byline=0&portrait=0"
    title="GitHub Universe: Operation Code"
    width="755"
  />
</div>
`;
